### PR TITLE
Update AndroidX Fragment to 1.3.4

### PR DESCRIPTION
**Pull request by TacoTheDank**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
- Updated AndroidX Fragment to 1.3.4 ([changelog](https://developer.android.com/jetpack/androidx/releases/fragment))
- Fix most deprecations from pre-1.3.0 Fragment versions
- Fix some of the onActivityResult deprecations

#### Fixes the following issue(s)
- Comes with whatever bugfixes were made between 1.2.4 and 1.3.4 (which are quite a lot). I highly recommend thoroughly reading the changelog.

#### APK testing 
On the website the APK can be found by going to the "Checks" tab below the title and then on "artifacts" on the right.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

## Notes
There are other deprecations that come with this upgrade that will need fixing at some point:
- ViewPager1 is now officially deprecated (YEAH BOIIIIII)

I didn't do other deprecations like onRequestPermissionsResult and the other onActivityResult ones because I'm not too confident and I want to keep this one smaller. I'll do them in other PRs.)

### CODE REVIEW

**Comment by TacoTheDank** [comment]
Also, I'm not that great at naming things, so feel free to tell me if you have better names in mind for the objects and functions and I'll rename them.

File: app/src/main/java/org/schabi/newpipe/MainActivity.java
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/MainActivity.java:603]: Shouldn't these also be migrated to the new way of handling permission results introduced in `1.3.0`? (also `RouterActivity.java`)

File: app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt:90]: Great!

File: app/build.gradle
Comment (Stypox) [inline app/build.gradle:201]: Why did you need to add a new library?

File: app/src/main/java/org/schabi/newpipe/BaseFragment.java
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/BaseFragment.java:125]: I'm not quite sure I understand this change. The behaviour here has changed.

**Review by Stypox — CHANGES_REQUESTED** [review]
I've read the changelog, but I do not know the architecture of fragments well enough to understand the changelog and your changes thoroughly.

(btw, do you happen to have some up-to-date and complete guide to fragments? I couldn't find good-quality material about it)

File: app/src/main/java/org/schabi/newpipe/MainActivity.java
Comment (TacoTheDank) [reply app/src/main/java/org/schabi/newpipe/MainActivity.java:603]: Yeah, but I'm going to do it in another PR, as mentioned above :)

File: app/build.gradle
Comment (TacoTheDank) [reply app/build.gradle:201]: We were only using a transitive version of the Fragment library, and that version was 1.2.4. We need to specify using Fragment version 1.3.4 to support the ActivityResult APIs.

File: app/src/main/java/org/schabi/newpipe/BaseFragment.java
Comment (TacoTheDank) [reply app/src/main/java/org/schabi/newpipe/BaseFragment.java:125]: Now that I think about it, I'm not quite sure myself. I'll revert this change.

File: app/src/main/java/org/schabi/newpipe/MainActivity.java
Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/MainActivity.java:603]: Oh, yes, sorry ;-)

File: app/src/main/java/org/schabi/newpipe/BaseFragment.java
Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/BaseFragment.java]: Yeah, I don't understand either what exactly this piece of code does. It was added in 8dde25532a06908c813b8b9909dc7d95f455960d, which doesn't provide any explanation. `getFragmentManager()` is just a deprecated `null`able alias of `getParentFragmentManager()`, isn't it? In another PR I think you should therefore probably replace `getFM()` everywhere with just `getParentFragmentManager()`, as it seems like just a workaround for the `null`ability.

**Review by Stypox — APPROVED** [review]
LGTM, thanks :-D

**Comment by TobiGr** [comment]
Did someone test this on KitKat? The app crashes constantly after trying to use the search once.

<details><summary>stracktrace: search</summary>
```
06-18 15:07:55.443 5829-5829/org.schabi.newpipe.debug E/ACRA: ACRA caught a NullPointerException for org.schabi.newpipe.debug
    java.lang.NullPointerException
        at org.schabi.newpipe.fragments.list.search.SearchFragment.onCreateOptionsMenu(SearchFragment.java:430)
        at androidx.fragment.app.Fragment.performCreateOptionsMenu(Fragment.java:3100)
        at androidx.fragment.app.FragmentManager.dispatchCreateOptionsMenu(FragmentManager.java:3181)
        at androidx.fragment.app.FragmentController.dispatchCreateOptionsMenu(FragmentController.java:391)
        at androidx.fragment.app.FragmentActivity.onCreatePanelMenu(FragmentActivity.java:288)
        at androidx.appcompat.view.WindowCallbackWrapper.onCreatePanelMenu(WindowCallbackWrapper.java:94)
        at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.onCreatePanelMenu(AppCompatDelegateImpl.java:3070)
        at androidx.appcompat.view.WindowCallbackWrapper.onCreatePanelMenu(WindowCallbackWrapper.java:94)
        at androidx.appcompat.app.ToolbarActionBar.populateOptionsMenu(ToolbarActionBar.java:456)
        at androidx.appcompat.app.ToolbarActionBar$1.run(ToolbarActionBar.java:57)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:761)
        at android.view.Choreographer.doCallbacks(Choreographer.java:574)
        at android.view.Choreographer.doFrame(Choreographer.java:543)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:747)
        at android.os.Handler.handleCallback(Handler.java:733)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:136)
        at android.app.ActivityThread.main(ActivityThread.java:5017)
        at java.lang.reflect.Method.invokeNative(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:515)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
        at dalvik.system.NativeStart.main(Native Method)
```
</details>

**Comment by TacoTheDank** [comment]
@TobiGr Are you sure that's related to this PR specifically? I didn't touch anything related to the search function...

**Comment by opusforlife2** [comment]
Seems like it: #6522. It specifically refers to your PR.

**Comment by TacoTheDank** [comment]
@opusforlife2 He may just be referring to 0.21.5. As of right now, Stypox's merge of my PR is the latest commit. Guess we'll have to wait for a response 🤔

**Comment by Douile** [comment]
Just to clear things up, it looks like #6522 is caused in some way by this PR: when building on commit 6e377dd3c552b059127e5a26ba2794c64e5f3892 the crash no longer occurs. lmk if you need more details about build or whatever.

**Comment by Douile** [comment]
I don't really understand this stuff and I don't see any mentions of changes to Fragment lifecycle here (https://developer.android.com/guide/fragments/lifecycle) but I noticed in SearchFragment.java service is only set in onResume. Copying the exact code from onResume (
```java
try {
    service = NewPipe.getService(serviceId);
} catch (final Exception e) {
    ErrorActivity.reportUiErrorInSnackbar(this,
            "Getting service for id " + serviceId, e);
}
```
) to onCreate means the crash (#6522) in SearchFragment no longer occurs. idk whether we still need it in onResume or whether this could cause problems in other fragments so I will leave that to you guys.

**Pull request merged**
